File: api-auth/src/ApiAuth.Application/Common/ApiResponse.cs
namespace ApiAuth.Application.Common;

public record ApiResponse<T>
{
    public required bool Success { get; init; }
    public string? Message { get; init; }
    public string? Code { get; init; }
    public T? Data { get; init; }

    public static ApiResponse<T> Ok(T data, string? message = null) =>
        new() { Success = true, Message = message, Data = data };

    public static ApiResponse<T> Fail(string message, T? data = default, string? code = null) =>
        new() { Success = false, Message = message, Data = data, Code = code };
}

File: api-endosos/src/common/responses/api-response.ts
export class ApiResponse<T> {
  success: boolean;
  message: string | null;
  code: string | null;
  data: T | null;

  static ok<T>(data: T | null, message: string | null = null): ApiResponse<T> {
    return Object.assign(new ApiResponse<T>(), { success: true, message, code: null, data });
  }

  static fail<T>(message: string, data: T | null = null, code: string | null = null): ApiResponse<T> {
    return Object.assign(new ApiResponse<T>(), { success: false, message, code, data });
  }
}
